tests/CanadaPost/Tests/NCShipmentTest.php

```diff
@@ -0,0 +1,313 @@
+            . '/../Mocks/ncshipment-response-success.xml');
+        $mock = new MockHandler([
+            new Response(200, [], $body),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $shipment = $this->shipmentService->createNCShipment(
+            $this->mockAddress(),
+            $this->mockAddress(),
+            $this->mockParcel(),
+            ['handler' => $handler]
+        );
+
+        $this->checkShippingResponse($shipment);
+    }
+    /**
+     * Test the GET request to retrieve an artifact from Canada Post.
+     *
+     * @covers NCShipment::getNCShipment()
+     * @test
+     */
+    public function getNCShipment()
+    {
+        $getShipmentBody = file_get_contents(__DIR__
+            . '/../Mocks/ncshipment-response-success.xml');
+        $getDetailsBody = file_get_contents(__DIR__
+            . '/../Mocks/ncshipment-details-success.xml');
+        $getReceiptBody = file_get_contents(__DIR__
+            . '/../Mocks/ncshipment-receipt-success.xml');
+        $mock = new MockHandler([
+            new Response(200, [], $getShipmentBody),
+            new Response(200, [], $getDetailsBody),
+            new Response(200, [], $getReceiptBody),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $shipment = $this->shipmentService->getNCShipment('0123456789', '', ['handler' => $handler]);
+        $details = $this->shipmentService->getNCShipment('0123456789', '', ['handler' => $handler]);
+        $receipt = $this->shipmentService->getNCShipment('0123456789', '', ['handler' => $handler]);
+
+        $this->checkShippingResponse($shipment);
+        $this->checkShippingDetails($details);
+        $this->checkShippingReceipt($receipt);
+    }
+
+    /**
+     * Test the GET request to retrieve an artifact from Canada Post.
+     *
+     * @covers NCShipment::getNCShipments()
+     * @test
+     */
+    public function getNCShipments()
+    {
+        $body = file_get_contents(__DIR__
+            . '/../Mocks/ncshipments-response-success.xml');
+        $mock = new MockHandler([
+            new Response(200, [], $body),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $shipments = $this->shipmentService->getNCShipments('', '', '', ['handler' => $handler]);
+
+        // Check response.
+        $this->assertTrue(is_array($shipments['non-contract-shipments']));
+
+        // Test the shipments.
+        $links = $shipments['non-contract-shipments']['link'];
+
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787234',
+            $links[0]['@attributes']['href']);
+
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787352',
+            $links[1]['@attributes']['href']);
+
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787123',
+            $links[2]['@attributes']['href']);
```

Comment:
It's probably GitHub breaking these into 2 lines, you should be doing so yourself:

```
$this->assertEquals(
  'https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787123',
  $links[2]['@attributes']['href']
);
```